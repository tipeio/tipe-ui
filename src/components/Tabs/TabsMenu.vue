<template>
  <div :data-tipe-ui="$options.name">
    <tipe-tabs-menu-button
      v-for="(button, i) in buttons"
      :key="i"
      v-bind="button"
      @click="$emit('click', button.label)"
    />
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import interfaces from '@/interfaces'
import TipeTabsMenuButton from './TabsMenuButton'

export default {
  name: 'TipeTabsMenu',
  components: {
    TipeTabsMenuButton
  },
  props: {
    buttons: vueTypes.arrayOf(vueTypes.shape(interfaces.tabsMenuButton))
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeTabsMenu'] {
  height: 2.6875rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: left;
  border-bottom: 1px solid var(--gray-light);
  box-sizing: border-box;
}
</style>
