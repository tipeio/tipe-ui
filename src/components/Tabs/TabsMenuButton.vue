<template>
  <button
    :class="rootClassObject"
    :data-tipe-ui="$options.name"
    @click="$emit('click')"
  >
    <div :class="layoutClassObject">
      <div
        v-if="icon"
        class="icon-container"
      >
        <tipe-icon :icon="icon"/>
      </div>
      <div class="label">
        {{ label }}
      </div>
    </div>
  </button>
</template>

<script>
import TabsMenuButtonShape from '@/types/TabsMenuButton'
import TipeIcon from '@/components/Icon'

export default {
  name: 'TipeTabsMenuButton',
  components: {
    TipeIcon
  },
  props: TabsMenuButtonShape,
  computed: {
    rootClassObject() {
      return {
        active: this.active
      }
    },
    layoutClassObject() {
      return {
        'icon-layout': this.icon,
        'label-layout': !this.icon
      }
    }
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeTabsMenuButton'] {
  height: 2.6875rem;
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background-color: transparent;
  box-sizing: border-box;
  border-bottom: 0.1875rem solid transparent;
}

.label-layout {
  display: flex;
  align-items: center;
  padding-left: 0.8125rem;
  box-sizing: border-box;
}

.icon-layout {
  display: grid;
  grid-template-columns: 2rem auto;
  align-items: center;
}

.icon-container {
  display: flex;
  justify-content: center;
}

.label {
  display: flex;
  justify-content: flex-start;
  font-size: 0.8125rem;
  letter-spacing: 0.02rem;
  color: var(--text-gray);
  box-sizing: border-box;
  padding-right: 2.5rem;
}

.active {
  border-bottom: 0.1875rem solid var(--purple);
}
</style>
