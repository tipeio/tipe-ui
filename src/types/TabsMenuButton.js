import vueTypes from 'vue-types'

export default {
  icon: vueTypes.string.def(''),
  label: vueTypes.string.def(''),
  active: vueTypes.bool.def(false)
}
