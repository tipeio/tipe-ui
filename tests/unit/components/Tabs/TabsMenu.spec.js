import { shallowMount } from '@vue/test-utils'
import TabsMenu from '@/components/Tabs/TabsMenu.vue'
import TabsMenuButton from '@/components/Tabs/TabsMenuButton.vue'
import tabsMenuButtonMock from '@/mocks/TabsMenuButton'
import createManyMocks from '@/mocks/createManyMocks'

describe('TabsMenu', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(TabsMenu)

      expect(wrapper).toMatchSnapshot()
    })

    it('<root> - has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(TabsMenu)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeTabsMenu')
    })
  })

  describe(':props', () => {
    it(':buttons - renders a TabsMenuButton component for each button', () => {
      const propsData = { buttons: createManyMocks(tabsMenuButtonMock, 3) }
      const wrapper = shallowMount(TabsMenu, { propsData })

      expect(wrapper.findAll(TabsMenuButton)).toHaveLength(3)
    })

    it(':buttons.label - should pass label prop to TabsMenuButton component', () => {
      const propsData = { buttons: [tabsMenuButtonMock()] }
      const wrapper = shallowMount(TabsMenu, { propsData })

      expect(wrapper.find(TabsMenuButton).props().label).toEqual(
        propsData.buttons[0].label
      )
    })

    it(':buttons.icon - should pass icon prop to TabsMenuButton component', () => {
      const propsData = { buttons: [tabsMenuButtonMock()] }
      const wrapper = shallowMount(TabsMenu, { propsData })

      expect(wrapper.find(TabsMenuButton).props().icon).toEqual(
        propsData.buttons[0].icon
      )
    })

    it(':buttons.active - should pass active prop to TabsMenuButton component', () => {
      const propsData = { buttons: [tabsMenuButtonMock()] }
      const wrapper = shallowMount(TabsMenu, { propsData })

      expect(wrapper.find(TabsMenuButton).props().active).toEqual(
        propsData.buttons[0].active
      )
    })
  })
})
