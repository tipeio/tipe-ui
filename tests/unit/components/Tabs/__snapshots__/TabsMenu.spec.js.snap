// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsMenu <template> matches previous snapshot 1`] = `Promise {}`;
